src/components/vm-and-vnic-list.ts

```diff
@@ -0,0 +1,113 @@
+import * as paper from 'paper';
+import { VmData, VmComponent } from './vm';
+import { VnicComponent } from './vnic';
+import { CONNECTOR_RADIUS, CONNECTOR_SIZE, CONNECTOR_RIGHT_PADDING, DEFAULT_STROKE_WIDTH, VAPP_NETWORK_RIGHT_MARGIN,
+  LABEL_HEIGHT, VM_MARGIN_VERTICAL } from '../constants/dimensions';
+import { VappNetworkPositionsByName } from './vapp-network-list';
+
+const VAPP_NETWORK_PADDING_RIGHT = 4.5;
+
+/**
+ * Interface for the lowest vnic point by network name.
+ */
+export interface LowestVnicPointByNetworkName {
+  [name: string]: paper.Point;
+}
+
+/**
+ * Vm List Visual Component.
+ */
+export class VmAndVnicListComponent extends paper.Group {
+
+  // store lowest vnic point by network name for setting the lowest point of the vapp network path
+  private _lowestVnicPointByNetworkName: LowestVnicPointByNetworkName = {};
+  // x position of the last (furthest right) network in vapp network list used for positioning vms and vnics
+  readonly lastNetworkPositionX: number = 0;
+
+  /**
+   * Creates a new VmAndVnicListComponent instance.
+   *
+   * @param _vms the vms data
+   * @param vappNetworkPositionsByName the vapp network positions by name from the VappNetworkListComponent for
+   * positioning x value of matching vnics
+   * @param _point the location that the vm and vnic list should be rendered at
+   */
+  constructor(private _vms: Array<VmData>,
+              private vappNetworkPositionsByName: VappNetworkPositionsByName = {},
+              private _point: paper.Point = new paper.Point(0, 0)) {
+    super();
+    this.applyMatrix = false;
+    this.pivot = new paper.Point(0, 0);
+    this.position = _point;
+
+    // reusable calculations
+    const vappNetworkCount = Object.keys(this.vappNetworkPositionsByName).length;
+    this.lastNetworkPositionX = vappNetworkCount && (vappNetworkCount - 1) * VAPP_NETWORK_RIGHT_MARGIN
+      + CONNECTOR_RADIUS - DEFAULT_STROKE_WIDTH;
```

Comment:
could also loop through vappNetworkPositionsByName from line 36 to find max position instead of calculating it like this. does it matter?

Reply:
actually, there's an even simpler way of getting it from the bounds of the VappNetworkListComponent from the main VappComponent.

Reply:
yeah, probably better to get it that way vs. having this calculation that is dependent on other, previous calulations